sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/AgentPoolData.java

```diff
@@ -0,0 +1,497 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.containerservice.models;
+
+import com.azure.resourcemanager.containerservice.fluent.models.AgentPoolInner;
+import com.azure.resourcemanager.resources.fluentcore.arm.ResourceUtils;
+import com.azure.resourcemanager.resources.fluentcore.model.HasInnerModel;
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * The client-side data of an agent pool.
+ */
+public class AgentPoolData implements AgentPool, HasInnerModel<AgentPoolInner> {
```

Comment:
Any suggestion for the name? AgentPoolRequest? AgentPoolContent?

Reply:
AgentPoolData sounds ok to me. But if we want another class in `implementation` package to return to the user, we may call this `AgentPoolRequest` and that in `implementation` `AgentPoolData`?

Reply:
Anything in `implementation` package should not affect naming of things in exposed models (implementation is not supposed to be exposed, and you can even name the subclass same `AgentPoolData` or `MonkeyData` in implementation and still go on with it)